### ISSUE TITLE
bug: False positive on react-you-might-not-need-an-effect/no-event-handler

### ISSUE THREAD

**Issue by reteps** [NickvanDyke/eslint-plugin-react-you-might-not-need-an-effect#25]
### Did you check docs and existing issues?

- [x] I have read the plugin docs
- [x] I have searched the existing issues
- [x] I have read [You Might Not Need An Effect](https://react.dev/learn/you-might-not-need-an-effect) and am reasonably confident that I *do* need an effect

### Plugin version

latest

### ESLint version

laest

### Describe the bug

On re-render (e.g. you hit 'next' on a step, and want the newly-rendered UI to auto-focus an input), we want to focus certain elements. Is there another idiomatic way to do this? Otherwise, I would consider this a false positive.

```ts
useEffect(() => {
    // eslint-disable-next-line react-you-might-not-need-an-effect/no-event-handler
    if (step === 2) {
      confirmationTextRef.current?.focus();
    }
  }, [step]);
```

### Steps To Reproduce

-

### Expected Behavior

Not flagged, since it is `.focus()`

**Comment by NickvanDyke** [comment]
Hmm yeah I guess you have to wait for re-render and can't call `focus()` right where you call `setStep(2)`. Good point!

Does it focus correctly if you put this inside `useLayoutEffect`? Currently the plugin analyzes those too, but after seeing this example, maybe it shouldn't. When used correctly, `useLayoutEffect` is used to interact with the DOM = external system = the effect is (most likely) valid.

I think `useLayoutEffect` is more idiomatic here too? But let me know if that's wrong.

**Comment by reteps** [comment]
Yes, thank you for the advice. I think one of those solutions will work.

**Comment by NickvanDyke** [comment]
Great, thanks! I updated the plugin in `v0.4.2` to not analyze `useLayoutEffect`s because they should generally be valid, and I imagine not every case will have a more specific solution like yours and `autofocus`.

Thanks for the report!
